src/OpenTelemetry.Api/Trace/Tracer.cs

```diff
@@ -58,14 +58,14 @@ public TelemetrySpan CurrentSpan
         /// </summary>
         /// <param name="name">Span name.</param>
         /// <param name="kind">Kind.</param>
-        /// <param name="attributes">Initial attributes for the span.</param>
+        /// <param name="initialAttributes">Initial attributes for the span.</param>
         /// <param name="links"> <see cref="Link"/> for the span.</param>
         /// <param name="startTime"> Start time for the span.</param>
         /// <returns>Span instance.</returns>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public TelemetrySpan StartRootSpan(string name, SpanKind kind = SpanKind.Internal, IEnumerable<KeyValuePair<string, object>> attributes = null, IEnumerable<Link> links = null, DateTimeOffset startTime = default)
+        public TelemetrySpan StartRootSpan(string name, SpanKind kind = SpanKind.Internal, ActivityTagsCollection initialAttributes = null, IEnumerable<Link> links = null, DateTimeOffset startTime = default)
```

Comment:
since this is a shim, should we create a shim around ActivityTagsCollection, and name it SpanAttributes or something like that? So users of this shim, continue to fully operate on OTel terms of Span and SpanAttribute.

Reply:
In the end, if we create the SpanAttributes it would be almost the same things as IEnumerable, right? So, maybe this change isn't needed. What do u think?

Reply:
Its not simply IEnumerable. It follows the rules listed here:
https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/common/common.md#attributes

The shim can only chose to support what is supported by OTel Spec - namely primitive types and array of them.

Reply:
What do you think about this:
```csharp
    public struct SpanAttribute
    {
        private readonly List<KeyValuePair<string, object>> attributes;

        public SpanAttribute()
        {
            this.attributes = new List<KeyValuePair<string, object>>();
        }

        public void Add(string key, int value)
        {
            this.attributes.Add(new KeyValuePair<string, object>(key, value));
        }

// add all accepted 
}
```

Reply:
This should be written on top of ActivityTagsCollection, and shouldn't have anything other than ActivityTagsCollection instane. i.e no `List<KeyValuePair<string, object>> attributes`

Reply:
What about:
```csharp
public struct SpanAttribute
    {
        private readonly ActivityTagsCollection attributes;

        public SpanAttribute()
        {
            this.attributes = new ActivityTagsCollection();
        }

        public void Add(string key, int value)
        {
            this.attributes.Add(key, value);
        }
    }
```
So, right now i'm just using the TagsCollection. Still not complete, but the idea is to expose only the accepted primitives and array of primitives

Reply:
Yes this looks good for me.

Reply:
Just updated. Can you validate? Thank you.